Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -86,10 +86,10 @@ namespace AzToolsFramework
             // Copies the instance DOM, that is stored in the focused or root template DOM, into the output instance DOM.
             AZStd::string relativePathFromTop = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances);
             PrefabDomPath relativeDomPath(relativePathFromTop.c_str());
-            PrefabDom focusedOrRootTemplateDom;
-            focusedOrRootTemplateDom.CopyFrom(m_prefabSystemComponentInterface->FindTemplateDom(focusedOrRootInstance->GetTemplateId()),
+            PrefabDom focusedOrRootTemplateDomCopy;
+            focusedOrRootTemplateDomCopy.CopyFrom(m_prefabSystemComponentInterface->FindTemplateDom(focusedOrRootInstance->GetTemplateId()),
                 instanceDom.GetAllocator());
-            const PrefabDomValue* instanceDomFromTemplate = relativeDomPath.Get(focusedOrRootTemplateDom);
+            const PrefabDomValue* instanceDomFromTemplate = relativeDomPath.Get(focusedOrRootTemplateDomCopy);
```

Comment:
Following this pattern, should we also wrap instanceDomFromTemplate in local scope? I see some other cases in this file that retrieve pointers to a dom that aren't wrapped in local scope.

Reply:
that's right. I thought about whether I add it or not since it is a pointer to a copied DOM. Eventually I decided to add "copy" to the variable name instead. I think we can keep it for now.

These few lines will need be updated to make a copy of the instance DOM rather than the entire template DOM. It will look like the code [L92-104 in this branch](https://github.com/o3de/o3de/commit/4177f668de029200d936939ecdd8bcb70db4f74e#diff-46548f9a25a132e1ae9f72ce7b62e390a8c01310025a3584b2afabf333dfe1b9). So at that time I will add a single scope so the pattern is followed.

Thanks for bringing this up!